From: Andreas Noever <andreas.noever@gmail.com>
Date: Fri, 20 Jun 2014 21:42:24 +0200
Subject: [26/31] thunderbolt: Make enum tb_drom_entry_type unsigned
Origin: https://git.kernel.org/linus/e7120778a4518a1c8f188ef9865058f7f5a36919

Force enum tb_drom_entry_type to unsigned to fix the following error:

drivers/thunderbolt/eeprom.c:202:39: error: dubious one-bit signed bitfield

Signed-off-by: Andreas Noever <andreas.noever@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/thunderbolt/eeprom.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-3.16.7/drivers/thunderbolt/eeprom.c
===================================================================
--- linux-3.16.7.orig/drivers/thunderbolt/eeprom.c
+++ linux-3.16.7/drivers/thunderbolt/eeprom.c
@@ -191,7 +191,8 @@ struct tb_drom_header {
 } __packed;
 
 enum tb_drom_entry_type {
-	TB_DROM_ENTRY_GENERIC,
+	/* force unsigned to prevent "one-bit signed bitfield" warning */
+	TB_DROM_ENTRY_GENERIC = 0U,
 	TB_DROM_ENTRY_PORT,
 };
 
